### PR TITLE
import symbol under cursor

### PR DESCRIPTION
**Pull request by n00dl3**
I added the possibility to import the symbol currently under the cursor (ctrl+shift+m when the cursor is on the symbol).
It's not a clean way to do this (I changed a method from private to public, etc). But I didn't have time to dive deeply in the code. AFAIK, It works and I find this functionality very interesting. So maybe it would give you some ideas to implement this in a cleaner manner.

Thank you for this plugin.

### CODE REVIEW

**Comment by buehler** [comment]
Oh nice one :-)
I'll review the code right away

File: src/extensions/ResolveExtension.ts [hunk 1]
```diff
@@ -123,7 +124,31 @@ export class ResolveExtension extends BaseExtension {
             }
         });
     }
-
+    private addImportFromCursor() {
+        if (!this.cache.cacheReady) {
+            this.showCacheWarning();
+            return;
+        }
+        console.log(vscode.window.activeTextEditor.selection);
```
Comment (buehler) [inline src/extensions/ResolveExtension.ts]: please remove the console.log

File: src/extensions/ResolveExtension.ts [hunk 2]
```diff
@@ -123,7 +124,31 @@ export class ResolveExtension extends BaseExtension {
...
+        let selection = vscode.window.activeTextEditor.selection;
+        let document = vscode.window.activeTextEditor.document;
+        let word = document.getWordRangeAtPosition(selection.active);
+        let searchText: string;
```
Comment (buehler) [inline src/extensions/ResolveExtension.ts]: could you use a variable statement list?
i.e.
```
let a = 'foo',
    b = 'bar';
```

File: src/provider/ResolveQuickPickProvider.ts [hunk 3]
```diff
@@ -14,14 +14,19 @@ export class ResolveQuickPickProvider {
...
     }
 
-    private buildQuickPickList(openDocument: vscode.Uri, openSource: string): Thenable<ResolveQuickPickItem[]> {
+    public buildQuickPickList(openDocument: vscode.Uri, openSource: string, search: string = ''): Thenable<ResolveQuickPickItem[]> {
         return this.parser.parseSource(openSource)
             .then(parsedSource => {
                 let resolveItems = this.resolveItemFactory.getResolvableItems(this.cache.cachedFiles, openDocument, parsedSource.imports);
+                if (search) {
+                    resolveItems = resolveItems.filter(item => {
+                        return item.declaration.name === search;
```
Comment (buehler) [inline src/provider/ResolveQuickPickProvider.ts]: this lambda can be simplified with `item => item.declaration.name === search`

**Comment by buehler** [comment]
I got some comments, after this, I'll merge the PR

Cheers

**Comment by n00dl3** [comment]
done

**Comment by buehler** [comment]
@n00dl3 Sorry, you missed the `console.log` @ https://github.com/buehler/typescript-hero/pull/24#discussion-diff-77473658R132
I don't have write access to your branch, so I cannot inline-edit the line ;-)

**Comment by n00dl3** [comment]
:rage4:

**Pull request merged**